nlp/2021_11_25_gpt3mix/nlp_gpt3mix.ipynb

```diff
@@ -0,0 +1,2575 @@
+            "padding": null,
+            "grid_auto_rows": null,
+            "grid_gap": null,
+            "max_width": null,
+            "order": null,
+            "_view_module_version": "1.2.0",
+            "grid_template_areas": null,
+            "object_position": null,
+            "object_fit": null,
+            "grid_auto_columns": null,
+            "margin": null,
+            "display": null,
+            "left": null
+          }
+        }
+      }
+    }
+  },
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "LXro9efVU84k"
+      },
+      "source": [
+        "# 📈: Text Augmentation using GPT-3\n",
+        "\n",
+        "This notebook looks into the possibility of performing data augmentation on an NLP dataset using the GPT-3 language model.\n",
+        "\n",
+        "Data augmentation techniques are used to generate additional samples. Data augmentation is already standard practice in computer vision projects 👌, but can also be leveraged in many NLP problems. We'll use a limited training set to simulate a real-world use case, where we often are constrained by the size of the available data 🤦."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "acwqGIAyU84m"
+      },
+      "source": [
+        "## 🛠️ Getting started\n",
+        "\n",
+        "The cells below will setup everything that is required to get started with data augmentation and finetuning an NLP model with the HuggingFace API."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "buIDjl_wU84m"
+      },
+      "source": [
+        "### Setup"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "v-ItvjqP4Cxn",
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "outputId": "c7b549a9-2c36-4a22-abc9-154973fdb0d5"
+      },
+      "source": [
+        "!!pip install -qq transformers datasets tokenizers openai requests"
+      ],
+      "execution_count": null,
+      "outputs": [
+        {
+          "output_type": "execute_result",
+          "data": {
+            "text/plain": [
+              "[]"
+            ]
+          },
+          "metadata": {},
+          "execution_count": 1
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Ir6zD__FU84n"
+      },
+      "source": [
+        "### Imports"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "hM-09Chsj7wI"
+      },
+      "source": [
+        "import re\n",
+        "import json\n",
+        "import torch\n",
+        "import random\n",
+        "import requests\n",
+        "import numpy as np\n",
+        "import pandas as pd \n",
+        "import plotly.express as px\n",
+        "import plotly.graph_objects as go\n",
+        "\n",
+        "from plotly.subplots import make_subplots\n",
```

Comment:
tip: better to group library imports together: so you can put all 3 `plotly` imports under one another